src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,243 @@
+    assertEquals(
+        expectedTermRatingEntity.getProperty("perception-class"),
+        termRatingQuery.getProperty("perception-class"));
+    
+    expectedTermRatingEntity.setProperty("hours", Long.valueOf(10));
+    assertEquals(
+        expectedTermRatingEntity.getProperty("hours"), termRatingQuery.getProperty("hours"));
+    
+    expectedTermRatingEntity.setProperty("difficulty", Long.valueOf(5));
+    assertEquals(
+        expectedTermRatingEntity.getProperty("difficulty"),
+        termRatingQuery.getProperty("difficulty"));
+    
+    expectedTermRatingEntity.setProperty("comments-professor", "The professor was still okay.");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("comments-professor"),
+        termRatingQuery.getProperty("comments-professor"));
+    
+    expectedTermRatingEntity.setProperty("perception-professor", Long.valueOf(3));
+    assertEquals(
+        expectedTermRatingEntity.getProperty("perception-professor"),
+        termRatingQuery.getProperty("perception-professor"));
+  }
+
+  @Test
+  public void AddingTermRatingWithSentimentAnalysis() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("class-input")).thenReturn("I do not like this.");
+    when(request.getParameter("rating-class")).thenReturn("4");
+    when(request.getParameter("hoursOfWork")).thenReturn("8");
+    when(request.getParameter("difficulty")).thenReturn("4");
+    when(request.getParameter("prof-input")).thenReturn("The professor was amazing.");
+    when(request.getParameter("rating-professor")).thenReturn("3");
+    when(request.getParameter("languages")).thenReturn("false");
+    when(request.getParameter("ID")).thenReturn("numberOneId");
+    Entity termEntity = new Entity("Term");
+    Key termKey = termEntity.getKey();
+    when(request.getParameter("CourseClass"))
+        .thenReturn(Course("Spring 2020", "Smith", 12, termKey));
+
+    newTermRating.addTermRating(request);
+
+    Entity termRatingQuery =
+        queryEntities("Rating", "reviewer-id", request.getParameter("reviewer-id")).get(0);
+    Entity expectedTermRatingEntity = new Entity("Rating");
+    expectedTermRatingEntity.setProperty("score-class", -0.8999999761581421);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("score-class"),
+        termRatingQuery.getProperty("score-class"));
+    
+    expectedTermRatingEntity.setProperty("score-professor", 0.8999999761581421);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("score-professor"),
+        termRatingQuery.getProperty("score-professor"));
+  }
+
+  @Test
+  public void AddingTermRatingWithTranslation() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("class-input")).thenReturn("Esta clase es muy interesante.");
```

Comment:
This is probably alright for now but it's not unlikely that this test will break later since you're relying on the translation service to always behave the same way (which I doubt it will in all cases). More ideal is to mock this out since it's not directly under test. (You should usually avoid mocking things you don't need to but since this is technically non-deterministic as far as you're concerned then it makes since in this case.)

Reply:
Oh that makes sense, I will look into it but when you mean "mock this out" in terms of translation what specifically are you referring to?

Reply:
Mock out the response from the translateService